lambda_function.py

```diff
@@ -1,9 +1,161 @@
+                return response["StackSetOperation"]["StackSetId"]
+
+            logger.info(f"Current operation status {response['StackSetOperation'].get('Status')}, going to sleep for {self.STACK_OPERATION_WAIT_SLEEP}")
+            time.sleep(self.STACK_OPERATION_WAIT_SLEEP)
+
+    def delete_stack_instances(self) -> Dict:
+        logger.info(f"Deleting stack instance for StackSet: {self.MASTER_ACCOUNT_PERMISSIONS_STACK_SET_NAME}, "
+                    f"region: {self.region_name}, AccountId: {self.customer_account_id}")
+
+        response = self.cloudformation_client.delete_stack_instances(StackSetName=self.MASTER_ACCOUNT_PERMISSIONS_STACK_SET_NAME,
+                                                                Regions=[self.region_name], RetainStacks=True,
+                                                                Accounts=[self.customer_account_id])
+        return response
+
+    def delete_stack_set(self):
+        logger.info(f"Deleting StackSet: {self.MASTER_ACCOUNT_PERMISSIONS_STACK_SET_NAME}")
+        return self.cloudformation_client.delete_stack_set(StackSetName=self.MASTER_ACCOUNT_PERMISSIONS_STACK_SET_NAME)
+
+    def register_to_dome9(self) -> Dict:
+        dome9_client = Client()
+        credentials = CloudAccountCredentials(arn=self.customer_account_new_role_arn, secret=self.customer_account_external_id)
+        payload = CloudAccount(name=self.customer_account_name, credentials=credentials)
+        resp = dome9_client.aws_cloud_account.create(body=payload)
+        return resp
+
+    def create_stack_set_flow(self) -> None:
+        try:
+            self.create_stack_set()
+        except Exception as e:
+            if "NameAlreadyExistsException" in repr(e):
+                logger.info(f"Received NameAlreadyExistsException for stack_set '{self.MASTER_ACCOUNT_PERMISSIONS_STACK_SET_NAME}'. Error: {repr(e)}")
+                self.delete_stack_instances()
+                time.sleep(20)
+                #self.delete_stack_set()
+                #time.sleep(30)
+                #self.create_stack_set()
+            else:
+                raise
+
+    def execute_onboarding_flow(self):
+        self.create_stack_set_flow()
+        self.create_stack_instances()
+        time.sleep(20)
+        register_result = self.register_to_dome9()
+        return register_result
 
 
 def lambda_handler(event, context):
-	print("Received event: " + json.dumps(event, indent=2))
-	return event  # Echo back the first key value
-	#raise Exception('Something went wrong')
+    logger.info(f"Event: {event}")
+    logger.info(f"Context: {context}")
+
+    event = event["detail"]
+
+    logger.info(f"Event reported state: {event['serviceEventDetails']['createManagedAccountStatus']['state']}'")
+    onboarder = Onboarder(event["awsRegion"], event["serviceEventDetails"]["createManagedAccountStatus"]["account"]["accountId"], event["serviceEventDetails"]["createManagedAccountStatus"]["account"]["accountName"])
```

Comment:
extract the dict values to separate parameters.
`account_id = event["serviceEventDetails"]["createManagedAccountStatus"]["account"]["accountId"]`

Reply:
done